### PR TITLE
Recover from pod restarts during cluster creation during setup

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
During testing of [Issue 476](https://github.com/canonical/mysql-k8s-operator/issues/476), we encountered a case where the leader unit pod that is creating the cluster exits with code 137 before it can successfully create the cluster. The pod then gets rescheduled, but is unable to recover from the conditions, as full cluster crash recovery is only done if all units are `offline` (which is not the case because there are 2 units in `waiting` to join cluster)  

## Solution
Attempt to `reboot_cluster_from_complete_outage` in above scenario. If the reboot from complete outage fails in above scenario, we should be safe to `drop_metadata_schema()` on unit whose pod was rescheduled, and recreate the cluster.
Added integration test

## TODO
Update the shared mysql charm lib code in the vm repo

### CODE REVIEW

File: lib/charms/mysql/v0/mysql.py [hunk 1]
```diff
@@ -619,6 +619,18 @@ def cluster_initialized(self) -> bool:
 
         return False
 
+    @property
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply lib/charms/mysql/v0/mysql.py:626]: updated libpatch - will make sure that these changes get propagated to the libs in the vm repo and published correctly

File: lib/charms/mysql/v0/mysql.py [hunk 2]
```diff
@@ -619,6 +619,18 @@ def cluster_initialized(self) -> bool:
...
+    def total_cluster_node_count(self) -> int:
+        """Get the cluster node counts spanning the entire cluster."""
+        if not self.app_peer_data.get("cluster-name"):
+            return 0
+
+        total_cluster_nodes = 0
+        for unit in self.app_units:
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply lib/charms/mysql/v0/mysql.py:633]: renamed, and refactored to be more precise in [f0daa96](https://github.com/canonical/mysql-k8s-operator/pull/499/commits/f0daa96819bd8af69af1075da2995f9d59dbd158) (with the property renamed in a following commit)

File: tests/integration/high_availability/test_crash_during_setup.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        await write_content_to_file_in_unit(
+            ops_test,
+            unit,
+            f"/var/lib/juju/agents/unit-{unit_label}/charm/disable",
+            "",
+            container_name="charm",
+        )
+
+    logger.info("Deleting pod")
+    delete_pod(ops_test, leader_unit)
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/high_availability/test_crash_during_setup.py:74]: will add in a follow up PR

**Review by taurus-forever — APPROVED** [review]
:crossed_fingers:

**Pull request merged**